compiler/scopelang/from_desugared.ml

```diff
@@ -863,7 +863,7 @@ let translate_program
                   match states with
                   | D.WholeVar -> WholeVar (ScopeVar.fresh (var_name, var_pos))
                   | States states ->
-                    let var_prefix = var_name ^ "_" in
+                    let var_prefix = var_name ^ "#" in
```

Comment:
Are we sure this gets correctly changed back to `_` in the C or Python backends?

Reply:
Yes, that would otherwise be a bug!

Reply:
Re: indeed, there was a bug where this would appear in the backend language (but it was a symptom of a wider bug where we would cleanup unicode characters but not non-ident characters that were part of ASCII like `é`).
The latest patch fixes that for all backends (just fixing the normalisation functions used by the `Renaming` module).